### PR TITLE
Added Doxyfile

### PR DESCRIPTION
**Pull request by ahcorde**
Added Doxyfile to include this package as part of the ROS 2 public API, available at docs.ros2.org. 

Signed-off-by: ahcorde <ahcorde@gmail.com>

### CODE REVIEW

**Comment by thomas-moulard** [comment]
@ahcorde How did you validate this is working?

**Comment by ahcorde** [comment]
you can just simply run inside the `libstatistics_collector` package and the open in a browser `doc_output/html/index.html` :

```
doxygen Doxyfile
```

But I'm integrating many packages in this repo https://github.com/ros2/docs.ros2.org which it's the one on charge of generate the documentation. PR to the repo is on the way

**Comment by ahcorde** [comment]
I don't have write permissions here. can you merge it?

**Comment by thomas-moulard** [comment]
> you can just simply run inside the `libstatistics_collector` package and the open in a browser `doc_output/html/index.html` :
> 
> ```
> doxygen Doxyfile
> ```
> 
> But I'm integrating many packages in this repo https://github.com/ros2/docs.ros2.org which it's the one on charge of generate the documentation. PR to the repo is on the way

Thanks, I'll try that! Do you have any doc about how to maintain this file? I'd like to understand when/how we should update it? Seems like it's only including the README.md now?

File: Doxyfile [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+ENABLE_PREPROCESSING   = YES
+MACRO_EXPANSION        = YES
+EXPAND_ONLY_PREDEF     = YES
+PREDEFINED             += LIBSTATISTICS_COLLECTOR_PUBLIC=
+
+# Tag files that do not exist will produce a warning and cross-project linking will not work.
+TAGFILES += "../../../../doxygen_tag_files/cppreference-doxygen-web.tag.xml=http://en.cppreference.com/w/"
+TAGFILES += "../../../../doxygen_tag_files/rcl.tag=http://docs.ros2.org/latest/api/rcl/"
+TAGFILES += "../../../../doxygen_tag_files/rcpputils.tag=http://docs.ros2.org/latest/api/rcpputils/"
```
Comment (thomas-moulard) [inline Doxyfile:25]: When should we change those?

File: Doxyfile [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+# Uncomment to generate tag files for cross-project linking.
+GENERATE_TAGFILE = "../../../../doxygen_tag_files/libstatistics_collector.tag"
```
Comment (thomas-moulard) [inline Doxyfile:27]: Ditto here?

**Comment by thomas-moulard** [comment]
Merging as this worked, @ahcorde if you could teach us (or write some doc?) about what we're expected to do with this file in the future, that'd be great.

**Comment by dabonnie** [comment]
@thomas-moulard this could be interesting, I'll take a look at it later: https://github.com/marketplace/actions/doxygen-action

**Comment by thomas-moulard** [comment]
Nice! Let's see what @ahcorde come up with. That should be fairly similar to `docs.ros2.org`. Maybe we could use the action in PRs though to ensure the Doxyfile is still valid?

**Comment by dabonnie** [comment]
>Maybe we could use the action in PRs though to ensure the Doxyfile is still valid?

My thought exactly. We would at least have local generation to manually review changes, etc.

**Comment by ahcorde** [comment]
Sorry for the late response, 

Doxygen and Sphinx are generated using this [script](https://github.com/ros2/docs.ros2.org/blob/doc_gen/build_docs.sh). We manually run it right now. Probably it makes sense to create a package similar to this one https://github.com/ros-infrastructure/rosdoc_lite

**Pull request merged**